tools/icu/icutrim.py

```diff
@@ -285,7 +285,7 @@ def addTreeByType(tree, mytree):
         # read in the resource list for the tree
         treelistfile = os.path.join(options.tmpdir,"%s.lst" % tree)
         runcmd("iculslocs", "-i %s -N %s -T %s -l > %s" % (outfile, dataname, tree, treelistfile))
-        fi = open(treelistfile, 'rb')
+        fi = open(treelistfile, 'r')
```

Comment:
That change is unlikely to produce the right results unless the default locale is ASCII or UTF-8. You can probably do this:
```python
try:
  fi = open(treelistfile, 'r', encoding='utf-8')  # py3
except TypeError:
  fi = open(treelistfile, 'rb')  # py2, no 'encoding' kw argument
```

Reply:
Is the use of `io.open()` the preferred solution?

Reply:
Oh, good point. Yes, `io.open(treelistfile, encoding='utf-8')` should work with both versions.

Reply:
thanks for your help. I followed your suggestions